internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -443,24 +450,35 @@ func (c *enrollCmd) prepareFleetTLS() error {
 
 func (c *enrollCmd) daemonReloadWithBackoff(ctx context.Context) error {
```

Comment:
I wonder if it would be possible to simplify the implementation of this function a bit.

Right now, we call `c.daemonReload(ctx)`, do some error checking, and then enter the backoff loop.  Inside the backoff loop we again call `c.daemonReload(ctx)` and do some error checking.  Would it be possible to consolidate and directly call the backoff loop, with one extra iteration and moving the `backExp.Wait()` to the end of each iteration instead of the start?

Reply:
It would be even better if the retry implementation took care of the loop and called a `func() error` on its own till success or deadline/timeout/max attempts. We could separate the concerns of retry from the actual action being performed